p2p/conn/connection.go

```diff
@@ -315,7 +337,18 @@ FOR_LOOP:
 			c.Logger.Debug("Send Ping")
 			wire.WriteByte(packetTypePing, c.bufWriter, &n, &err)
 			c.sendMonitor.Update(int(n))
+			c.Logger.Debug("Starting pong timer", "dur", c.config.PongTimeout)
+			c.pongTimer = time.AfterFunc(c.config.PongTimeout, func() {
+				c.pongTimeoutCh <- true
+			})
 			c.flush()
+		case timeout := <-c.pongTimeoutCh:
+			if timeout {
+				c.Logger.Debug("Pong timeout")
+				err = errors.New("pong timeout")
+			} else {
+				c.stopPongTimer()
```

Comment:
weird that we stop it here... i don't think this is the right place to be stopping it.

Reply:
also, see my first comment way up.

Reply:
weird?) we need to stop it here to prevent pongTimer from firing (`c.pongTimeoutCh <- true` above). 

```
pongTimerStart ............. (got pong -> stop pongTimer) xxx (pongTimerFires) ..... (pingTimer fires)
pongTimerStart ............. (pongTimer fires)
```